include/fiction/algorithms/simulation/sidb/quickexact.hpp

```diff
@@ -0,0 +1,477 @@
+            {
+                charge_lyt.assign_defect(
+                    cell, sidb_defect{defect.type, defect.charge, charge_lyt.get_phys_params().epsilon_r});
+            }
+            else if (defect.epsilon_r != 0 && defect.lambda_tf == 0)
+            {
+                charge_lyt.assign_defect(cell, sidb_defect{defect.type, defect.charge, defect.epsilon_r,
+                                                           charge_lyt.get_phys_params().lambda_tf});
+            }
+            else
+            {
+                charge_lyt.assign_defect(cell, defect);
+            }
+        }
+
+        charge_lyt.set_local_external_potential(parameter.local_external_potential);
+        charge_lyt.set_global_external_potential(parameter.global_potential, false);
+        detected_negative_sidb_indices = charge_lyt.negative_sidb_detection();
+        detected_negative_sidbs.reserve(detected_negative_sidb_indices.size());
+        all_sidbs_in_lyt_without_negative_detected_ones = charge_lyt.get_sidb_order();
+        real_placed_defects                             = charge_lyt.get_defects();
+        number_of_SiDBs                                 = charge_lyt.get_sidb_order().size();
+    }
+    /**
+     * This function is used to generate a layout without the SiDBs that are detected to be negatively charged in a
+     * physically-valid layout.
+     */
+    void generate_layout_without_negative_sidbs()
+    {
+        for (const auto& index : detected_negative_sidb_indices)
+        {
+            const auto cell = charge_lyt.index_to_cell(static_cast<uint64_t>(index));
+            detected_negative_sidbs.push_back(cell);
+            layout.assign_cell_type(cell, Lyt::cell_type::EMPTY);
+        }
+
+        // All detected negatively charged SiDBs are erased from the all_sidbs_in_lyt_without_negative_detected_ones
+        // vector.
+        all_sidbs_in_lyt_without_negative_detected_ones.erase(
+            std::remove_if(all_sidbs_in_lyt_without_negative_detected_ones.begin(),
+                           all_sidbs_in_lyt_without_negative_detected_ones.end(),
+                           [this](const typename Lyt::cell& n)
+                           {
+                               return std::find(detected_negative_sidbs.begin(), detected_negative_sidbs.end(), n) !=
+                                      detected_negative_sidbs.end();
+                           }),
+            all_sidbs_in_lyt_without_negative_detected_ones.end());
+    };
+    /**
+     * Cell-level layout
+     */
+    Lyt layout;
+    /**
+     * Charge distribution surface
+     */
+    charge_distribution_surface<Lyt> charge_lyt{};
+    /**
+     * Parameter used for the simulation.
+     */
+    quickexact_params<Lyt> parameter{};
+    /**
+     * Indices of all SiDBs that are detected to be negatively charged in a physically valid layout.
+     */
+    std::vector<int64_t> detected_negative_sidb_indices{};
```

Comment:
I see what you mean. I chose it because we are detecting SiDBs that must be negatively charged. However, I agree that this could be misleading. I will change it.